### PR TITLE
ReturnExecutor New style

### PR DESCRIPTION
**Pull request by maierlars**

### CODE REVIEW

File: arangod/Aql/ReturnExecutor.cpp [hunk 1]
```diff
@@ -45,3 +45,66 @@ ReturnExecutor::ReturnExecutor(Fetcher& fetcher, ReturnExecutorInfos& infos)
...
+    TRI_ASSERT(input.isInitialized());
+    // REMARK: it is called `getInputRegisterId` here but FilterExecutor calls it `getInputRegister`.
+    AqlValue val = input.stealValue(_infos.getInputRegisterId());
+    AqlValueGuard guard(val, true);
+    TRI_IF_FAILURE("ReturnBlock::getSome") {
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
+    }
+    output.moveValueInto(_infos.getOutputRegisterId(), input, guard);
+    output.advanceRow();
+    stats.incrCounted();
```
Comment (goedderz) [inline arangod/Aql/ReturnExecutor.cpp]: You removed the condition `if (_infos.doCount())` for this line; was that intentional, or should that read
```suggestion
  if (_infos.doCount()) {
    stats.incrCounted();
  }
```
? I'm in favour of removing such special cases, if it works.

File: arangod/Aql/ReturnExecutor.h [hunk 2]
```diff
@@ -94,38 +94,18 @@ class ReturnExecutor {
...
+
+  /**
+   * @brief produce the next Rows of Aql Values.
+   *
+   * @return ExecutorState, the stats, and a new Call that needs to be send to upstream
+   */
+  [[nodiscard]] auto produceRows(AqlItemBlockInputRange& input, OutputAqlItemRow& output)
+      -> std::tuple<ExecutorState, Stats, AqlCall>;
+
+  auto expectedNumberOfRows(size_t atMost) const
```
Comment (goedderz) [inline arangod/Aql/ReturnExecutor.h]:
```suggestion
  [[nodiscard]] auto expectedNumberOfRows(size_t atMost) const
```

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8377/

**Comment by mchacki** [comment]
Additions:
* Implemented skipRowsRange (not invoked in produce path)
* Added tests

New Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8438/

**Comment by mchacki** [comment]
Above Jenkins found in issue in another Executor.
Fixed now, new Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8456/

**Comment by mchacki** [comment]
Failing test now blue \o/

**Pull request merged**